src/services/identity/SitesCacheService.ts

```diff
@@ -14,81 +18,232 @@ import type { GitHubRepositoryData, RepositoryData } from "@root/types/repoInfo"
 // Changes are polled at fixed intervals, stored in cache, and served
 // to avoid long load time from live querying.
 
+const DEFAULT_PAGE_PARAMS = {
+  per_page: GH_MAX_REPO_COUNT,
+  sort: "full_name",
+  page: 1,
+}
+
+type GitHubResponseHeaders = HeadersDefaults & {
+  link?: string
+}
+
+type Link = {
+  relid: "first" | "last" | "prev" | "next"
+  url: string
+  pagenum: number
+}
+
+type LinkSet = {
+  first?: Link
+  last?: Link
+  prev?: Link
+  next?: Link
+}
+
+type LinkMatch = RegExpExecArray & {
+  groups: {
+    relid: "first" | "last" | "prev" | "next"
+    url: string
+  }
+}
+
+type FetchRepoPageResult = {
+  repos: RepositoryData[]
+  links?: LinkSet
+}
+
+type CacheStore = {
+  [key: string]: RepositoryData
+}
+
+function parseGitHubLinkHeader(linkheader: string) {
+  // example value: link: <https://api.github.com/organizations/40887764/repos?page=2>; rel="next", <https://api.github.com/organizations/40887764/repos?page=34>; rel="last"
+  const links: LinkSet = {}
+
+  const linkRe = /<(?<url>[^>]+)>; rel="(?<relid>[a-z]+)"(, )?/g
+  let regRes: LinkMatch | null = null
+
+  // eslint-disable-next-line no-cond-assign
+  while ((regRes = linkRe.exec(linkheader) as LinkMatch) !== null) {
```

Comment:
style: i think this is a little bit confusing, what do you think about just calling `linkRe.exec(linkheader)` on the `while` then assigning inside the loop? 

```suggestion
  while (linkRe.exec(linkheader) !== null) {
    const regRes = linkRe.exec(linkheader) as LinkMatch
```

Reply:
No that would be wrong. Because `.exec()` traverses the input, consecutive calls to `.exec()` return different matches (i.e. not idempotent). Duplicating the calls would miss some of input matches.

It's a typical pattern of using regexes that is not supported by this annoying eslint rule 😑

Reply:
Hmm, That said I suppose using the "newer" API [`String.matchAll()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll) might work better here 🤔

Reply:
I pushed a commit to use `String.MatchAll()` instead. Let me know if you find it more readable. 

Sadly I still needed to add a eslint disable rule because a tight loop over an iterator is considered dangerous, but in this case, we know the input set is limited (max 4 links), so it's fine (I added a comment for that)